R/value.R

```diff
@@ -107,3 +107,275 @@ inspect_na_value <- function(value) {
   return(success())
 
 }
+
+
+
+#' Check if the passed entity is a single floating point which is (strictly)
+#' greater than another specified value.
+#'
+#' @param value the value to check
+#' @param comparator the minimum allowed value, inclusive or exclusive.
+#' @param strictly If TRUE (default), the comparator value is checked exclusive
+#' \code{(>)}. If FALSE the comparator value is checked inclusive \code{(>=)}.
+#'
+#' @param allow_na if true, accept a value that is NA.
+#' @param allow_null if true, accept a value that is NULL.
+#'
+#' @examples
+#' \dontrun{
+#' # For assertion
+#' assertthat::assert_that(qscheck::is_greater_than_value(value, comparator))
+#' # For check
+#' if (qscheck::is_greater_than_value(value, comparator)) {}
+#' }
+#'
+#' @concept value
+#' @export
+is_greater_than_value <- function(
+  value, comparator, strictly = TRUE,
+  allow_na = FALSE, allow_null = FALSE
+) {
+
+  res <- inspect_greater_than_value(
+    value, comparator,
+    strictly = strictly,
+    allow_na = allow_na,
+    allow_null = allow_null
+  )
+
+  return(res$valid)
+}
+assertthat::on_failure(is_greater_than_value) <- function(call, env) {
+  value <- callget(call, env, "value", NULL)
+  comparator <- callget(call, env, "comparator", NULL)
+  strictly <- callget(call, env, "strictly", TRUE)
+  allow_na <- callget(call, env, "allow_na", FALSE)
+  allow_null <- callget(call, env, "allow_null", FALSE)
+
+  res <- inspect_greater_than_value(
+    value, comparator,
+    strictly = strictly,
+    allow_na = allow_na,
+    allow_null = allow_null
+  )
+
+if (is.null(value) || is.null(comparator)) {
+  return(paste(
+    res$reason
+  ))
+} else {
+  return(paste0(
+    call$value,
+    snippet_must_be(paste0("greater value than ", call$comparator)),
+    snippet_na(allow_na),
+    snippet_null(allow_null),
+    ". ", res$reason
+  ))
+}
+}
+inspect_greater_than_value <- function(
+  value, comparator, strictly = TRUE,
+  allow_na = FALSE, allow_null = FALSE
+) {
+
+  if (is.null(value)) {
+    if (allow_null == TRUE) {
+      return(success())
+    } else {
+      return(failure("Passed value is NULL"))
+    }
+  }
+
+  if (is.null(comparator)) {
+    return(failure("Passed comparator value is NULL"))
+  }
+
+  if (!is.numeric(value)) {
+    return(failure("Passed value is not a numerical"))
+  }
+
+  if (!is.numeric(comparator)) {
+    return(failure("Passed comparator value is not a numerical"))
+  }
+
+  if (length(value) != 1) {
+    return(failure(
+      "Passed value must be a single numerical value, not a vector"
+    ))
+  }
+
+  if (length(comparator) != 1) {
+    return(failure(
+      "Passed comparator value must be a single numerical value, not a vector"
+    ))
+  }
+
+  if (is_na_value(value)) {
```

Comment:
of you can use the is functions, but they don't return the reason, only true or false.

Also beware of possible recursive situations, in which an inspect function calls another inspect function that calls the first. It happened to me once.